xcdat/dataset.py

```diff
@@ -317,6 +319,13 @@ def decode_time(dataset: xr.Dataset) -> xr.Dataset:  # noqa: C901
     ds = dataset.copy()
     coord_keys = _get_all_coord_keys(ds, "T")
 
+    if len(coord_keys) == 0:
+        raise KeyError(
+            "No time coordinates were found in this dataset to decode. Make sure that "
+            "either the CF 'axis' or 'standard_name' attribute is set on time "
+            "coordinates and try decoding again."
+        )
+
```

Comment:
Note, the multi-file dataset used [here](https://github.com/xCDAT/xcdat/pull/343#discussion_r1005010637) was not actually being decoded in xCDAT (which might have influenced runtime results)

The time coordinates don't have CF attributes set so cf-xarray/xcdat can't find them. I added a catch here to make sure it raises a KeyError if no time coordinate keys were found.

Reply:
I referenced the wrong comment here.

The dataset that was not actually being decoded was `fn = "/p/css03/cmip5_css02/data/cmip5/output1/ICHEC/EC-EARTH/historical/mon/atmos/Amon/r1i1p1/tas/1/*.nc"`. I'm not sure where I found this dataset path, maybe it is buried in a comment somewhere.

Reply:
I am not sure if this is CF compliant...I get the following from [this site](https://pumatest.nerc.ac.uk/cgi-bin/cf-checker.pl):

> CHECKING NetCDF FILE: /tmp/15373.nc
> =====================
> Using CF Checker Version 4.1.0
> Checking against CF Version CF-1.4
> Using Standard Name Table Version 79 (2022-03-19T15:25:54Z)
> Using Area Type Table Version 10 (23 June 2020)
> Using Standardized Region Name Table Version 4 (18 December 2018)
> ...
> ------------------
> Checking variable: time
> ------------------
> WARN: (3): No standard_name or long_name attribute specified
> 
> ------------------
> Checking variable: time_bnds
> ------------------
> INFO: attribute calendar is being used in a non-standard way
> 
> ------------------
> Checking variable: tas
> ------------------
> INFO: attribute history is being used in a non-standard way
> 
> ERRORS detected: 0
> WARNINGS given: 1
> INFORMATION messages: 2

Reply:
What if we add something like the following in `_get_all_coord_keys`:

```
    if CF_ATTR_MAP['T']['coordinate'] in obj.coords.keys():
        keys = keys + [CF_ATTR_MAP['T']['coordinate']]
```

This would mean that if the standard name is listed as a coordinate, then we could include that as a key (even if it were missing the standard/long_name. 

I just think it would be a bummer if a coordinate axis is called `time`, but we can't recognize it as a time coordinate.